src/Language/PureScript/Errors.hs

```diff
@@ -845,11 +849,14 @@ rethrow f = flip catchError $ \e -> throwError (f e)
 -- Rethrow an error with source position information
 --
 rethrowWithPosition :: (MonadError MultipleErrors m) => SourceSpan -> m a -> m a
-rethrowWithPosition pos = rethrow (onErrorMessages withPosition)
-  where
-  withPosition :: ErrorMessage -> ErrorMessage
-  withPosition (PositionedError _ err) = withPosition err
-  withPosition err = PositionedError pos err
+rethrowWithPosition pos = rethrow (onErrorMessages (withPosition pos))
+
+warnWithPosition :: (MonadWriter MultipleErrors m) => SourceSpan -> m a -> m a
+warnWithPosition pos = censor (onErrorMessages (withPosition pos))
+
+withPosition :: SourceSpan -> ErrorMessage -> ErrorMessage
+withPosition pos (PositionedError _ err) = withPosition pos err
+withPosition pos err = PositionedError pos err
```

Comment:
Also, is this right? It's what we had before, but if we're going top down surely it would be better to keep the original position, as it will be more specific.